content_script.js

```diff
@@ -164,6 +145,87 @@ function log(message) {
 }
 
 function isCategoryPage() {
+  const currentURL = new URL(document.URL);
+  const pathArray = currentURL.pathname.split('/');
+  const baseURL = "https://api.swift.gg/content/";
+  const url = baseURL + pathArray[pathArray.length-2] + '/' + pathArray[pathArray.length-1];
+
   const lastPath = pathArray[pathArray.length - 1] || pathArray[pathArray.length - 2];
   return endUpWhiteList.includes(lastPath)
+}
+
+function addInstructionToCategoryPage() {
+  let contentDiv = document.getElementsByClassName("copy-container")[0]
+  let spaceElement = document.createElement("br");
+  let spaceElement2 = document.createElement("br");
+  let style = document.createElement("style");
+  style.innerHTML = "p.indicator {\n" +
+      "    font-size: 18px;\n" +
+      "    font-weight: bold;\n" +
+      "}";
+  document.head.appendChild(style);
+  let pElement = document.createElement("p")
+  pElement.classList.add("indicator")
+  pElement.textContent = "SwiftGG 正在运行，请点击上方按钮开始学习 ⬆️"
+  contentDiv.appendChild(spaceElement)
+  contentDiv.appendChild(spaceElement2)
+  contentDiv.appendChild(pElement)
+}
+chrome.runtime.onMessage.addListener(
+    function(request, sender, sendResponse) {
+      // listen for messages sent from background.js
+      if (request.message === pageSwitchedRequestMethod) {
+        if (request.url.includes("developer.apple.com")) {
+          chrome.runtime.sendMessage({type: initialRequestMethod}, (response) => {
+            log(`Flag status: ${response.shouldTranslate}`);
+
+            startTranslate(response.shouldTranslate)
+
+            log("Plugin wait page loaded");
+
+          });
+        }
+      }
+    }
+);
+
+function startTranslate(shouldTranslate) {
+  const currentURL = new URL(document.URL);
+  previousGlobalUrl = globalUrl
+  globalUrl = currentURL
+  globalUrl.search = ""
+  globalUrl.hash = ""
+  const pathArray = currentURL.pathname.split('/');
+  const baseURL = "https://api.swift.gg/content/";
```

Comment:
Please put the baseURL out of the function.

Reply:
If we put outside, we cannot fix the switching problem

Reply:
let's talk offline